File: C#/blackjack.cs
using System;
using System.Collections.Generic;

class Card
{
    public string Rank;
    public string Suit;

    public Card(string rank, string suit)
    {
        Rank = rank;
        Suit = suit;
    }

    public string Display()
    {
        return Rank + " " + Suit;
    }

    public int Value()
    {
        if (Rank == "J" || Rank == "Q" || Rank == "K")
        {
            return 10;
        }
        else if (Rank == "A")
        {
            return 11;
        }
        else
        {
            return int.Parse(Rank);
        }
    }
}

class Deck
{
    public List<Card> Cards = new List<Card>();
    Random random = new Random();

    public Deck()
    {
        string[] suits = { "Kier", "Karo", "Trefl", "Pik" };
        string[] ranks = { "A", "2", "3", "4", "5", "6", "7", "8", "9", "10", "J", "Q", "K" };

        for (int i = 0; i < suits.Length; i++)
        {
            for (int j = 0; j < ranks.Length; j++)
            {
                Cards.Add(new Card(ranks[j], suits[i]));
            }
        }
    }

    public void Shuffle()
    {
        for (int i = 0; i < Cards.Count; i++)
        {
            int randomIndex = random.Next(Cards.Count);

            Card temp = Cards[i];
            Cards[i] = Cards[randomIndex];
            Cards[randomIndex] = temp;
        }
    }

    public Card DrawCard()
    {
        Card card = Cards[0];
        Cards.RemoveAt(0);
        return card;
    }
}

class Hand
{
    public List<Card> Cards = new List<Card>();

    public void AddCard(Card card)
    {
        Cards.Add(card);
    }

    public int CalculateValue()
    {
        int total = 0;
        int aces = 0;

        foreach (Card card in Cards)
        {
            total += card.Value();

            if (card.Rank == "A")
            {
                aces++;
            }
        }

        while (total > 21 && aces > 0)
        {
            total -= 10;
            aces--;
        }

        return total;
    }

    public void Display()
    {
        foreach (Card card in Cards)
        {
            Console.WriteLine(card.Display());
        }

        Console.WriteLine("Wartość ręki: " + CalculateValue());
    }
}

class Player
{
    public string Name;
    public Hand Hand;

    public Player(string name)
    {
        Name = name;
        Hand = new Hand();
    }
}

class Dealer
{
    public Hand Hand;

    public Dealer()
    {
        Hand = new Hand();
    }
}

class Program
{
    static void Main(string[] args)
    {
        Deck deck = new Deck();
        deck.Shuffle();

        Player player = new Player("Gracz");
        Dealer dealer = new Dealer();

        player.Hand.AddCard(deck.DrawCard());
        player.Hand.AddCard(deck.DrawCard());

        dealer.Hand.AddCard(deck.DrawCard());
        dealer.Hand.AddCard(deck.DrawCard());

        Console.WriteLine("____BLACKJACK____");

        while (true)
        {
            Console.WriteLine("\nTwoje karty:");
            player.Hand.Display();

            if (player.Hand.CalculateValue() > 21)
            {
                Console.WriteLine("Przegrałeś. Masz więcej niż 21.");
                return;
            }

            Console.Write("Dobierasz kartę? hit/stand: ");
            string choice = Console.ReadLine();

            if (choice == "hit")
            {
                player.Hand.AddCard(deck.DrawCard());
            }
            else if (choice == "stand")
            {
                break;
            }
            else
            {
                Console.WriteLine("Wpisz hit albo stand.");
            }
        }

        Console.WriteLine("\nKarty dealera:");
        dealer.Hand.Display();

        while (dealer.Hand.CalculateValue() < 17)
        {
            Console.WriteLine("\nDealer dobiera kartę.");
            dealer.Hand.AddCard(deck.DrawCard());
            dealer.Hand.Display();
        }

        int playerValue = player.Hand.CalculateValue();
        int dealerValue = dealer.Hand.CalculateValue();

        Console.WriteLine("\n=== WYNIK ===");
        Console.WriteLine("Gracz: " + playerValue);
        Console.WriteLine("Dealer: " + dealerValue);

        if (dealerValue > 21)
        {
            Console.WriteLine("Wygrałeś. Dealer ma więcej niż 21.");
        }
        else if (playerValue > dealerValue)
        {
            Console.WriteLine("Wygrałeś.");
        }
        else if (playerValue < dealerValue)
        {
            Console.WriteLine("Przegrałeś.");
        }
        else
        {
            Console.WriteLine("Remis.");
        }
    }
}

File: Python/blackjack.py
import random


class Card:
    def __init__(self, rank, suit):
        self.rank = rank
        self.suit = suit

    def display(self):
        return self.rank + " " + self.suit

    def value(self):
        if self.rank in ["J", "Q", "K"]:
            return 10
        elif self.rank == "A":
            return 11
        else:
            return int(self.rank)


class Deck:
    def __init__(self):
        self.cards = []

        suits = ["Kier", "Karo", "Trefl", "Pik"]
        ranks = ["A", "2", "3", "4", "5", "6", "7", "8", "9", "10", "J", "Q", "K"]

        for suit in suits:
            for rank in ranks:
                self.cards.append(Card(rank, suit))

    def shuffle(self):
        random.shuffle(self.cards)

    def draw_card(self):
        return self.cards.pop()


class Hand:
    def __init__(self):
        self.cards = []

    def add_card(self, card):
        self.cards.append(card)

    def calculate_value(self):
        total = 0
        aces = 0

        for card in self.cards:
            total += card.value()

            if card.rank == "A":
                aces += 1

        while total > 21 and aces > 0:
            total -= 10
            aces -= 1

        return total

    def display(self):
        for card in self.cards:
            print(card.display())

        print("Wartość ręki:", self.calculate_value())


class Player:
    def __init__(self, name):
        self.name = name
        self.hand = Hand()


class Dealer:
    def __init__(self):
        self.hand = Hand()


def main():
    deck = Deck()
    deck.shuffle()

    player = Player("Gracz")
    dealer = Dealer()

    player.hand.add_card(deck.draw_card())
    player.hand.add_card(deck.draw_card())

    dealer.hand.add_card(deck.draw_card())
    dealer.hand.add_card(deck.draw_card())

    print("____BLACKJACK____")

    while True:
        print("\nTwoje karty:")
        player.hand.display()

        if player.hand.calculate_value() > 21:
            print("Przegrałeś. Masz więcej niż 21.")
            return

        choice = input("Dobierasz kartę? hit/stand: ")

        if choice == "hit":
            player.hand.add_card(deck.draw_card())
        elif choice == "stand":
            break
        else:
            print("Wpisz hit albo stand.")

    print("\nKarty dealera:")
    dealer.hand.display()

    while dealer.hand.calculate_value() < 17:
        print("\nDealer dobiera kartę.")
        dealer.hand.add_card(deck.draw_card())
        dealer.hand.display()

    player_value = player.hand.calculate_value()
    dealer_value = dealer.hand.calculate_value()

    print("\n=== WYNIK ===")
    print("Gracz:", player_value)
    print("Dealer:", dealer_value)

    if dealer_value > 21:
        print("Wygrałeś. Dealer ma więcej niż 21.")
    elif player_value > dealer_value:
        print("Wygrałeś.")
    elif player_value < dealer_value:
        print("Przegrałeś.")
    else:
        print("Remis.")


if __name__ == "__main__":
    main()
